{"name": "test_steps_demo[allure]", "status": "broken", "statusDetails": {"message": "FileNotFoundError: [Errno 2] No such file or directory: './result/b.png'", "trace": "test_data1 = 'allure'\n\n    @allure.testcase(\"http://www.github.com\")\n    @allure.feature(\"百度搜索\")\n    @pytest.mark.parametrize(\"test_data1\",yaml.safe_load(open('data.yml')))\n    def test_steps_demo(test_data1):\n        with allure.step(\"打开百度网页\"):\n            driver = webdriver.Chrome()\n            driver.get(\"http://www.baidu.com\")\n            driver.maximize_window()\n    \n        with allure.step(f'输入关键词：{test_data1}'):\n            driver.find_element_by_id(\"kw\").send_keys(test_data1)\n            time.sleep(2)\n            driver.find_element_by_id(\"su\").click()\n            time.sleep(2)\n    \n        with allure.step(\"保存图片\"):\n            driver.save_screenshot(\"./result/b.png\")\n>           allure.attach.file(\"./result/b.png\",attachment_type=allure.attachment_type.PNG)\n\ntest_baidudemo.py:25: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \ne:\\python36\\lib\\site-packages\\allure_commons\\_allure.py:167: in file\n    plugin_manager.hook.attach_file(source=source, name=name, attachment_type=attachment_type, extension=extension)\ne:\\python36\\lib\\site-packages\\pluggy\\hooks.py:286: in __call__\n    return self._hookexec(self, self.get_hookimpls(), kwargs)\ne:\\python36\\lib\\site-packages\\pluggy\\manager.py:93: in _hookexec\n    return self._inner_hookexec(hook, methods, kwargs)\ne:\\python36\\lib\\site-packages\\pluggy\\manager.py:87: in <lambda>\n    firstresult=hook.spec.opts.get(\"firstresult\") if hook.spec else False,\ne:\\python36\\lib\\site-packages\\allure_pytest\\listener.py:219: in attach_file\n    self.allure_logger.attach_file(uuid4(), source, name=name, attachment_type=attachment_type, extension=extension)\ne:\\python36\\lib\\site-packages\\allure_commons\\reporter.py:113: in attach_file\n    plugin_manager.hook.report_attached_file(source=source, file_name=file_name)\ne:\\python36\\lib\\site-packages\\pluggy\\hooks.py:286: in __call__\n    return self._hookexec(self, self.get_hookimpls(), kwargs)\ne:\\python36\\lib\\site-packages\\pluggy\\manager.py:93: in _hookexec\n    return self._inner_hookexec(hook, methods, kwargs)\ne:\\python36\\lib\\site-packages\\pluggy\\manager.py:87: in <lambda>\n    firstresult=hook.spec.opts.get(\"firstresult\") if hook.spec else False,\ne:\\python36\\lib\\site-packages\\allure_commons\\logger.py:53: in report_attached_file\n    shutil.copy2(source, destination)\ne:\\python36\\lib\\shutil.py:257: in copy2\n    copyfile(src, dst, follow_symlinks=follow_symlinks)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nsrc = './result/b.png'\ndst = 'F:\\\\hogwarts_gor\\\\python\\\\test_baidu\\\\results\\\\474f1d34-01fe-430b-b3a4-d551eb9e710c-attachment.png'\n\n    def copyfile(src, dst, *, follow_symlinks=True):\n        \"\"\"Copy data from src to dst.\n    \n        If follow_symlinks is not set and src is a symbolic link, a new\n        symlink will be created instead of copying the file it points to.\n    \n        \"\"\"\n        if _samefile(src, dst):\n            raise SameFileError(\"{!r} and {!r} are the same file\".format(src, dst))\n    \n        for fn in [src, dst]:\n            try:\n                st = os.stat(fn)\n            except OSError:\n                # File most likely does not exist\n                pass\n            else:\n                # XXX What about other special files? (sockets, devices...)\n                if stat.S_ISFIFO(st.st_mode):\n                    raise SpecialFileError(\"`%s` is a named pipe\" % fn)\n    \n        if not follow_symlinks and os.path.islink(src):\n            os.symlink(os.readlink(src), dst)\n        else:\n>           with open(src, 'rb') as fsrc:\nE           FileNotFoundError: [Errno 2] No such file or directory: './result/b.png'\n\ne:\\python36\\lib\\shutil.py:120: FileNotFoundError"}, "steps": [{"name": "打开百度网页", "status": "passed", "start": 1597995601060, "stop": 1597995607033}, {"name": "输入关键词：allure", "status": "passed", "start": 1597995607033, "stop": 1597995611194}, {"name": "保存图片", "status": "broken", "statusDetails": {"message": "FileNotFoundError: [Errno 2] No such file or directory: './result/b.png'\n", "trace": "  File \"F:\\hogwarts_gor\\python\\test_baidu\\test_baidudemo.py\", line 25, in test_steps_demo\n    allure.attach.file(\"./result/b.png\",attachment_type=allure.attachment_type.PNG)\n  File \"e:\\python36\\lib\\site-packages\\allure_commons\\_allure.py\", line 167, in file\n    plugin_manager.hook.attach_file(source=source, name=name, attachment_type=attachment_type, extension=extension)\n  File \"e:\\python36\\lib\\site-packages\\pluggy\\hooks.py\", line 286, in __call__\n    return self._hookexec(self, self.get_hookimpls(), kwargs)\n  File \"e:\\python36\\lib\\site-packages\\pluggy\\manager.py\", line 93, in _hookexec\n    return self._inner_hookexec(hook, methods, kwargs)\n  File \"e:\\python36\\lib\\site-packages\\pluggy\\manager.py\", line 87, in <lambda>\n    firstresult=hook.spec.opts.get(\"firstresult\") if hook.spec else False,\n  File \"e:\\python36\\lib\\site-packages\\pluggy\\callers.py\", line 208, in _multicall\n    return outcome.get_result()\n  File \"e:\\python36\\lib\\site-packages\\pluggy\\callers.py\", line 80, in get_result\n    raise ex[1].with_traceback(ex[2])\n  File \"e:\\python36\\lib\\site-packages\\pluggy\\callers.py\", line 187, in _multicall\n    res = hook_impl.function(*args)\n  File \"e:\\python36\\lib\\site-packages\\allure_pytest\\listener.py\", line 219, in attach_file\n    self.allure_logger.attach_file(uuid4(), source, name=name, attachment_type=attachment_type, extension=extension)\n  File \"e:\\python36\\lib\\site-packages\\allure_commons\\reporter.py\", line 113, in attach_file\n    plugin_manager.hook.report_attached_file(source=source, file_name=file_name)\n  File \"e:\\python36\\lib\\site-packages\\pluggy\\hooks.py\", line 286, in __call__\n    return self._hookexec(self, self.get_hookimpls(), kwargs)\n  File \"e:\\python36\\lib\\site-packages\\pluggy\\manager.py\", line 93, in _hookexec\n    return self._inner_hookexec(hook, methods, kwargs)\n  File \"e:\\python36\\lib\\site-packages\\pluggy\\manager.py\", line 87, in <lambda>\n    firstresult=hook.spec.opts.get(\"firstresult\") if hook.spec else False,\n  File \"e:\\python36\\lib\\site-packages\\pluggy\\callers.py\", line 208, in _multicall\n    return outcome.get_result()\n  File \"e:\\python36\\lib\\site-packages\\pluggy\\callers.py\", line 80, in get_result\n    raise ex[1].with_traceback(ex[2])\n  File \"e:\\python36\\lib\\site-packages\\pluggy\\callers.py\", line 187, in _multicall\n    res = hook_impl.function(*args)\n  File \"e:\\python36\\lib\\site-packages\\allure_commons\\logger.py\", line 53, in report_attached_file\n    shutil.copy2(source, destination)\n  File \"e:\\python36\\lib\\shutil.py\", line 257, in copy2\n    copyfile(src, dst, follow_symlinks=follow_symlinks)\n  File \"e:\\python36\\lib\\shutil.py\", line 120, in copyfile\n    with open(src, 'rb') as fsrc:\n"}, "attachments": [{"source": "474f1d34-01fe-430b-b3a4-d551eb9e710c-attachment.png", "type": "image/png"}], "start": 1597995611194, "stop": 1597995611581}], "parameters": [{"name": "test_data1", "value": "'allure'"}], "start": 1597995601060, "stop": 1597995611584, "uuid": "95d51b41-0add-4f5c-8e9b-4066f4a4f166", "historyId": "77446bc5c35b2e2394141393a0922721", "testCaseId": "c1bd71ed6b29aa553aa5af4cbca54344", "fullName": "test_baidudemo#test_steps_demo", "labels": [{"name": "feature", "value": "百度搜索"}, {"name": "suite", "value": "test_baidudemo"}, {"name": "host", "value": "LAPTOP-M1LUQJ0P"}, {"name": "thread", "value": "13148-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "test_baidudemo"}], "links": [{"type": "test_case", "url": "http://www.github.com", "name": "http://www.github.com"}]}